### ISSUE TITLE
No building image for already built building if use right mouse click

### ISSUE THREAD

**Issue by ihhub** [ihhub/fheroes2#110]
There is no building image for already built building if use right mouse click:
![image](https://user-images.githubusercontent.com/19829520/66053708-61fdee80-e565-11e9-85f8-679048c50710.png)

But in original game it looks like this:
![image](https://user-images.githubusercontent.com/19829520/66053772-7e9a2680-e565-11e9-9b38-ee21d80a859f.png)

**Comment by euronay** [comment]
I'll have a go at this tomorrow if it's OK?

**Comment by ihhub** [comment]
Sure!

**Comment by euronay** [comment]
When fixing this, I've noticed a couple of other things that are not quite the same.

- When you right click any building after you have built for the turn, it displays the "Cannot Build" dialog:

![image](https://user-images.githubusercontent.com/3864985/66124606-bbedcb00-e5dc-11e9-9506-6b04a4a76641.png)

But the original game shows the building with cost dialog:
![image](https://user-images.githubusercontent.com/3864985/66124633-d0ca5e80-e5dc-11e9-9e4c-b1b5f935b4ef.png)

- When you left click a building that cannot be built the same dialog is shown, but in the original game nothing happens.

I've changed these in my PR too - I hope this is OK?
